infrastructure/src/main/kotlin/detekt/DetektConfiguration.kt

```diff
@@ -0,0 +1,226 @@
+package com.redmadrobot.build.detekt
+
+import com.android.build.gradle.AppExtension
+import com.android.build.gradle.BaseExtension
+import com.android.build.gradle.LibraryExtension
+import com.android.build.gradle.TestExtension
+import com.android.build.gradle.internal.core.InternalBaseVariant
+import com.redmadrobot.build.DetektPlugin
+import com.redmadrobot.build.extension.RedmadrobotExtension
+import com.redmadrobot.build.internal.detekt.CollectGitDiffFilesTask
+import com.redmadrobot.build.internal.getFileIfExists
+import com.redmadrobot.build.internal.hasKotlinPlugin
+import com.redmadrobot.build.internal.hasPlugin
+import com.redmadrobot.build.internal.isInfrastructureRootProject
+import io.gitlab.arturbosch.detekt.Detekt
+import io.gitlab.arturbosch.detekt.DetektCreateBaselineTask
+import org.gradle.api.Project
+import org.gradle.api.tasks.SourceTask
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.kotlin.dsl.getByType
+import org.gradle.kotlin.dsl.named
+import org.gradle.kotlin.dsl.register
+
+internal class DetektConfiguration(
```

Comment:
Is it better than keeping these functions in `DetektPlugin`?
Plugin class now is nearly empty and all logic is still in one place here.

Reply:
I thought that the Detekt configuration project had become too complex, and Detekt reported a "TooManyFunctions" error. As I recall, we discussed this. But I also like to keep everything in one place. Moved the logic back.